apps/aecore/lib/aecore/account/account.ex

```diff
@@ -70,68 +102,32 @@ defmodule Aecore.Account.Account do
         ) :: {:ok, SignedTx.t()}
   def spend(sender, sender_priv_key, receiver, amount, fee, nonce) do
     payload = %{receiver: receiver, amount: amount}
-    spend_tx = DataTx.init(SpendTx, payload, sender, fee, nonce)
-    SignedTx.sign_tx(spend_tx, sender_priv_key)
+    spend_tx = DataTx.init(SpendTx, payload, [sender], fee, nonce)
+    SignedTx.sign_tx(spend_tx, sender, sender_priv_key)
   end
 
   @doc """
-  Adds balance to a given address (public key)
+  Adds balance to a given Account state and updates last update block.
   """
-  @spec transaction_in(Account.t(), integer()) :: Account.t()
-  def transaction_in(account_state, amount) do
+  @spec apply_transfer!(ChainState.account(), non_neg_integer(), integer()) ::
+          ChainState.account()
+  def apply_transfer!(account_state, block_height, amount) do
```

Comment:
I don't like the throw here. And I don't even see a reason for it. It is used in `process_chainstate` to check if the sender has enough coins to pay for the fees and the amount itself, but we check this in `preprocess_check`. That's the reason of the `preprocess_check` anyways.

Reply:
This is a sanity check. Better to have an unhanded error in block processing then to accept an invalid block.